### DISCUSSION TITLE
Add customizable colors

### DISCUSSION THREAD

**Discussion by glivolsi** [netbox-community/netbox#15024] [Ideas]
The actual color scheme have a poor colors palette, please add a customization feature to add custo colors

**Comment by jeremystretch** [comment]
NetBox provides 27 color options. These were specifically chosen because they are easy to distinguish from another. (IIRC we mostly reused colors from the Bootstrap library.)

The purpose of assigning a color to a value is to easily distinguish values via their color assignments. Allowing users to add arbitrary custom colors which are likely _not_ easily distinguished from one another would entirely defeat the purpose of color-coding. If the 27 colors available aren't sufficient for your needs, you shouldn't be relying on color coding in the first place.

**Reply by glivolsi** [reply]
I need multiple colors because I have many devices of the same type with different functionality, for example just for switches I have Access, Distribution, Core, Metro Edge, Dark Fiber Edge, TOR, etc. I would like to have all switches with various shades of Yellow but the current color scheme does not allow me to do that so I had to use the same color for different device role

![image](https://github.com/netbox-community/netbox/assets/5219941/639fb91c-e793-4b1d-8a46-a1a879957a50)

**Reply by jeremystretch** [reply]
You're not going to be able to readily distinguish between e.g. yellow-300 and yellow-500 in an arbitrary list of objects, so there's no reason to get so granular with color assignments. These gradients are nice to achieve aesthetic appeal for _design_ purposes, but not for categorization.

You should also bear in mind that many users have some degree of color deficiency, and expecting them to pick or recognize the exact correct color would be unreasonable.

**Reply by glivolsi** [reply]
Consider an optional feature, performing the merge between the 27 default colors and user-defined colors.

Anyway, congratulations on a fantastic product

**Reply by bradmoutoux** [reply]
This is my concern.  I use Colors and they are at times similar to distinguish same category, but slightly different specification.  For example, I record our Public and Private DNS in here.  I have Name Servers.  I have over 12 of them.  However, groups of 4 are by provider as our internal domain NS, GoDaddy, Oracle, and Azure.  I use blue for Internal, but I was a more bold blue for Azure as they are both Microsoft.

However, Netbox does have a workaround for this! Import.  If you Import a tag, you can give whatever HEX code you want (all lowercase) and it will import that tag with the proper color you want.  Caveat: You cannot change your tag aftewards as the edit form shows the color as not being there so you need to choose one of the predefined colors.

**Comment by LDTips** [comment]
I found that colors are customisable if you change them via REST API - there they can be set to any hex code.
Is there any possibility this funcionality will be available via UI? It's already supported on the back-end level
